src/mbgl/renderer/layers/render_hillshade_layer.cpp

```diff
@@ -65,14 +65,17 @@ bool RenderHillshadeLayer::hasCrossfade() const {
     return false;
 }
 
-void RenderHillshadeLayer::render(PaintParameters& parameters, RenderSource* src) {
+void RenderHillshadeLayer::prepare(const LayerPrepareParameters& params) {
+    RenderLayer::prepare(params);
+    if (auto* demsrc = params.source->as<RenderRasterDEMSource>()) {
+        maxzoom = demsrc->getMaxZoom();
```

Comment:
`assert(demsrc)`, old code had check for nullptr

Reply:
Checked at `if (auto* demsrc =`

Reply:
:+1: